src/options/smt_options.toml

```diff
@@ -173,6 +173,34 @@ header = "options/smt_options.h"
   read_only  = true
   help       = "check proofs eagerly with proof-new for local debugging"
 
+[[option]]
+  name       = "proofGranularityMode"
+  category   = "regular"
+  long       = "proof-granularity=MODE"
+  type       = "ProofGranularityMode"
+  default    = "THEORY_REWRITE"
+  help       = "modes for proof granularity"
+  help_mode  = "Modes for proof granularity."
+[[option.mode.OFF]]
+  name = "off"
+  help = "Do not improve the granularity of proofs."
+[[option.mode.REWRITE]]
+  name = "rewrite"
+  help = "Expand macros, but do not expand rewrite or substitution steps."
+[[option.mode.THEORY_REWRITE]]
+  name = "theory-rewrite"
+  help = "Expand macros, rewrite and substitution steps, but do not expand theory rewrite steps."
+[[option.mode.DSL_REWRITE]]
+  name = "dsl-rewrite"
+  help = "Expand all steps to the level of DSL rewrites and evaluation steps."
```

Comment:
This is a bit misleading naming scheme, since `rewrite` is the mode where we *do not* expand rewrites. Wouldn't it be better to have something like
- off
- macros
- rewrites-and-subs
- theory-rewrites

Reply:
Then I think the option should be called `proof-elim=`, since granularity refers to the smallest unit that is kept.  I think I prefer granularity still, perhaps the option description should be updated to make this clearer?

Reply:
I see, I got confused indeed. Improving the doc might clarify.